tests/test_framework/utils.py

```diff
@@ -338,36 +340,56 @@ def wrapper(*args, **kwargs):
     def call(self, method, payload=[]):
         self.logger.debug("Calling %s with payload %r", method, payload)
 
-        # FIXME: we open a new socket for every readobj call...
-        sock = UnixSocket(self.socket_path)
-        msg = json.dumps(
-            {
-                "jsonrpc": "2.0",
-                "id": 0,
-                "method": method,
-                "params": payload,
-            }
-        )
-        sock.sock.send(msg.encode())
-        this_id = self.next_id
-        resp = self._readobj(sock)
-
-        self.logger.debug("Received response for %s call: %r", method, resp)
-        if "id" in resp and resp["id"] != this_id:
-            raise ValueError(
-                "Malformed response, id is not {}: {}.".format(this_id, resp)
+        if platform.uname()[0] == "Windows":
+            bin = os.path.join(
+            os.path.dirname(__file__), "..", "..", "target", "debug", "revault-cli"
             )
-        sock.close()
-
-        if not isinstance(resp, dict):
-            raise ValueError(
-                "Malformed response, response is not a dictionary %s." % resp
+            cmd_line = [bin, "--conf", self.config_file, method, *payload]
+            output = subprocess.check_output(cmd_line, shell=True, universal_newlines=True, text=True, encoding='utf-8')
+            self.logger.debug(f"Got this output: {output}")
+            output = json.loads(output)
+            # output = bytes.decode("utf-8")
+            self.logger.debug("Received response for %s call: %r", method,output)
+
+            if not isinstance(output, dict):
+                raise ValueError(
+                    "Malformed response, response is not a dictionary %s." % output
+                )
+            elif "result" not in output:
+                raise ValueError('Malformed response, "result" missing.')   
+            return output["result"]
```

Comment:
All this is duplicated with the `else` branch. Can we rather unify it?

Reply:
Yep.